src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx

```diff
@@ -246,24 +351,31 @@ class DataPrivacyRulesPanelSelectorField extends React.Component<Props, State> {
           onFocus={this.handleFocus}
           disabled={disabled}
         />
-        {showSuggestions && suggestions.length > 0 && (
-          <SuggestionsWrapper ref={this.suggestionList}>
-            {suggestions.map((suggestion, index) => (
-              <SuggestionItem
-                key={suggestion.value}
-                onClick={this.handleClickSuggestionItem(suggestion)}
-                active={index === activeSuggestion}
-                tabIndex={-1}
-              >
-                <TextOverflow>{suggestion.value}</TextOverflow>
-                {suggestion?.description && (
-                  <SuggestionDescription>
-                    (<TextOverflow>{suggestion.description}</TextOverflow>)
-                  </SuggestionDescription>
-                )}
-              </SuggestionItem>
-            ))}
-          </SuggestionsWrapper>
+        {showSuggestions && (
+          <SuggestionsDropdown ref={this.suggestionList}>
+            <SuggestionsWrapper>
+              {suggestions.slice(0, 50).map((suggestion, index) => (
+                <SuggestionItem
+                  key={suggestion.value}
+                  onClick={this.handleClickSuggestionItem(suggestion)}
+                  active={index === activeSuggestion}
+                  tabIndex={-1}
+                >
+                  <TextOverflow>{suggestion.value}</TextOverflow>
+                  {suggestion?.description && (
+                    <SuggestionDescription>
+                      (<TextOverflow>{suggestion.description}</TextOverflow>)
+                    </SuggestionDescription>
+                  )}
+                </SuggestionItem>
+              ))}
+            </SuggestionsWrapper>
+            <EventIdTextField
+              name="eventId"
+              defaultValue={this.props.selectedEventId}
+              placeholder={t("Paste event ID for better suggestions")}
+              onChange={this.props.handleEventIdChange} />
```

Comment:
this as well

Reply:
when we pass method as prop, usually we name it with the prefix 'on', so in this case it would be `onEventIdChange`

Reply:
This is gone